### PR TITLE
2250: Manually generate interaction

### LINKED ISSUE #2250: Manually generate interaction
We have logic to generate interactions.

But, game-players are wary that an interaction may fall between the cracks - so they want to be able to generate one manually.

So, we need to add a new popup form to `AdjudicationMessagesList`.  The popup form will let the adjudicator provide some data.

The layout of the form is provided here: https://docs.google.com/presentation/d/185IQEG_yy3_16B3nDjqKWsGaigb2I54_IhqeVa0T-RY/edit#slide=id.g1d4daf1b3ae_0_0

This will be a collaborative piece of work:
- Ian can handle collating the data [1]
- Ian can provide the `Submit` handler [2]
- @lebaphi will create the form, please.  I _guess_ this will be something like the existing `CustomDialog`, but in the `helpers` sub-folder of `adjudication-messages-list`.

I don't really know how we do the form, or get the data in/out - so could to with @lebaphi  support on it.  Aah, hold on - I have a better understanding now.  The component will take these props:
```
export type Props = {
   // data to use to populate controls
   data: ManualInteractionData
   // close handler
   onClose: { (): void }
   // save handler, provides data from form controls
   onSave: { (data: ManualInteractionResults): void }
}
```

Note: I've started my bits, in #2251 

[1] See the `manualDialog` object
[2] See the `handleManualInteraction` method

### PR DESCRIPTION
**Pull request by lebaphi**
Fixes #2250

### CODE REVIEW

**Comment by lebaphi** [comment]
https://user-images.githubusercontent.com/26705621/212692826-50922d82-988d-4f18-885e-fde6f74b2a57.mov

File: packages/components/src/local/atoms/custom-dialog/index.tsx [hunk 1]
```diff
@@ -17,7 +17,8 @@ export const CustomDialog: React.FC<Props> = (props) => {
     bodyStyle,
     footerStyle,
     cancelBtnText = 'Cancel',
-    saveBtnText = 'Save'
+    saveBtnText = 'Save',
+    children
```
Comment (IanMayo) [inline packages/components/src/local/atoms/custom-dialog/index.tsx]: > introducing children prop

Great solution, well done @lebaphi

**Review by IanMayo — DISMISSED** [review]

**Comment by IanMayo** [comment]
great piece of work, thanks @lebaphi - will do some tidying tomorrow.

**Comment by lebaphi** [comment]
@IanMayo I think I need to clean up the custom dialog component a bit.

**Comment by IanMayo** [comment]
>  I think I need to clean up the custom dialog component a bit.

Thanks @lebaphi  - have you finished cleaning up?

**Comment by lebaphi** [comment]
> > I think I need to clean up the custom dialog component a bit.
> 
> Thanks @lebaphi - have you finished cleaning up?

Yes @IanMayo

**Pull request merged**